sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobClientBuilder.java

```diff
@@ -268,6 +330,7 @@ private HttpPipeline getHttpPipeline() {
         // Closest to API goes first, closest to wire goes last.
         List<HttpPipelinePolicy> policies = new ArrayList<>();
 
+        policies.add(new FetchEncryptionVersionPolicy(getUnencryptedBlobClient(), requiresEncryption));
```

Comment:
You're creating a whole extra blob client just for this policy?

Reply:
Yea. Clients are pretty lightweight, so I'm not especially worried about size. This may go away after preview as I'll probably need to override download methods in crypto for download to file, openInputStream, etc anyway. I may just go ahead and override all downloads.